### PR TITLE
cache curries to not repeat calls for concepts

### PR DESCRIPTION
**Pull request by Tom-Szendrey**
# Description
cache curries to not repeat calls for concepts

### CODE REVIEW

File: packages/client/hmi-client/src/services/concept.ts [hunk 1]
```diff
@@ -20,14 +21,29 @@ interface EntityMap {
...
+
+		// If a rename function is defined, loop over the first row
+		const response = await promise;
+		if (response?.status === 200 && response?.data) return response.data;
+		if (response?.status === 204) console.warn('No DKG entities found for curies:', curies);
+		return null;
+	} catch (err) {
+		logger.error(err);
+		return null;
+	}
```
Comment (mwdchang) [inline packages/client/hmi-client/src/services/concept.ts:46]: I think this setup will still get a ton of errors if the call fails - which goes into the catch block with the splendour of 100 toasters?

Comment (mwdchang) [reply packages/client/hmi-client/src/services/concept.ts:46]: Yeah we may need to tweak the axio interceptor to not spam toasters

**Review by mwdchang — APPROVED** [review]
The cache parts look fine, but the error handling will probably need to be rethought.

File: packages/client/hmi-client/src/services/concept.ts (hunk 1 above)
Comment (YohannParis) [reply packages/client/hmi-client/src/services/concept.ts:46]: Yes, spamming the user with 4XX and 5XX isn't great.

**Review by YohannParis — APPROVED** [review]
I would prefere to cache the `response.data` than the promise myself.

**Pull request merged**